internal/objects/reconciler.go

```diff
@@ -726,6 +728,16 @@ func (r *Reconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.U
 	case r.Mode == api.Remove:
 		return false
 
+	case r.Mode == api.Allow:
```

Comment:
I initially thought about implementing it in this way, but was unsure on what parameters to pass from the webhooks since `Validator` has no way of knowing the propagation mode? So what should I pass for the extra parameter from the webhook functions (for example `func (v *Validator) getConflictingObjectsOfType`)?

Reply:
What do you think? @adrianludwin

Reply:
The validator does have access to the propagation mode [here](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/347c5a3f92463d1157650172524d7743bdc6faa7/internal/objects/validator.go#L109) - is this what you mean?

Reply:
Might have missed it, I'll use the same method on the other validators, thanks